### PR TITLE
C4R-782 Practice Programs Search

### PR DESCRIPTION
**Pull request by dragos1195**
Disclaimer: Search is bugged, works on a second button push. WIP

https://wearetribus.atlassian.net/browse/C4R-782

### CODE REVIEW

File: src/common/components/filter-modal/FilterModal.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+                        }}
+                      />
+                      <Controller
+                        key={PracticeProgramsSearchConfig.start.key}
+                        name={PracticeProgramsSearchConfig.start.key}
+                        rules={PracticeProgramsSearchConfig.start.rules}
+                        control={control}
+                        render={({ field: { onChange, value } }) => {
+                          return (<DatePicker
+                            defaultValue={value ? value : undefined}
```
Comment (birloiflorian) [inline src/common/components/filter-modal/FilterModal.tsx:129]: Aici daca nu are value, nu vine ca undefined oricum?

File: src/common/components/filter-modal/FilterModal.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                        }}
+                      />
+                    </div>
+                    <div className="flex flex-col gap-2 w-full absolute bottom-4">
+                      <button
+                        type="button"
+                        className='flex bg-yellow w-full rounded font-titilliumSemiBold text-xl items-center justify-center p-3'
+                        onClick={handleSubmit(onApply)}
+                      >
+                        Aplica filtre
```
Comment (birloiflorian) [inline src/common/components/filter-modal/FilterModal.tsx:194]: Internationalizare

File: src/common/components/practice-programs-search/PracticeProgramsSearch.tsx [hunk 3]
```diff
@@ -19,58 +54,219 @@ const PracticeProgramsSearch = () => {
...
-                config={{
-                  ...PracticeProgramsSearchConfig.name.config,
-                  name: PracticeProgramsSearchConfig.name.key,
-                  error: errors[PracticeProgramsSearchConfig.name.key]?.message,
-                  defaultValue: value,
-                  onChange: onChange,
-                  id: 'search__term',
-                }}
+    <div className='bg-yellow w-full flex flex-col items-center px-2 sm:px-4 py-10 gap-8'>
+      <p className='font-titilliumBold sm:text-4xl text-xl  text-black'>Programe de practică la ONG-uri din România</p>
```
Comment (birloiflorian) [inline src/common/components/practice-programs-search/PracticeProgramsSearch.tsx:93]: Internationalizare

File: src/common/components/practice-programs-search/PracticeProgramsSearch.tsx [hunk 4]
```diff
@@ -19,58 +54,219 @@ const PracticeProgramsSearch = () => {
...
+          </div>
+        </div>
+        {props.showFilters && (
+          <div className='sm:hidden flex justify-flex-start h-14 items-center bg-white px-4 gap-2 rounded-md shadow w-fit cursor-pointer active:bg-gray-200'
+            onClick={() => setFilterModalOpen(true)}>
+            <p
+              id="create-organization-activity__button-back"
+              className="text-sm sm:text-base  h-full flex items-center"
+            >
+              Filtre
```
Comment (birloiflorian) [inline src/common/components/practice-programs-search/PracticeProgramsSearch.tsx:157]: Internationalizare

File: src/common/components/practice-programs-search/PracticeProgramsSearch.tsx [hunk 5]
```diff
@@ -19,58 +54,219 @@ const PracticeProgramsSearch = () => {
...
+            </p>
+            <AdjustmentsIcon className='w-5 h-5' />
+            <p
+              id="create-organization-activity__button-back"
+              className="text-base rounded-full bg-yellow p-2 flex items-center w-10 items-center justify-center"
+            >
+              5
```
Comment (birloiflorian) [inline src/common/components/practice-programs-search/PracticeProgramsSearch.tsx:164]: Aici nu ar trebui sa fie calculat?

File: src/common/components/practice-programs-search/PracticeProgramsSearch.tsx [hunk 6]
```diff
@@ -19,58 +54,219 @@ const PracticeProgramsSearch = () => {
...
+              );
+            }}
+          />
+          <button
+            id="create-organization-activity__button-back"
+            type="button"
+            className="text-sm sm:text-base text-yellow bg-black w-full h-full"
+            onClick={handleSubmit(search)}
+          >
+            Cauta
```
Comment (birloiflorian) [inline src/common/components/practice-programs-search/PracticeProgramsSearch.tsx:260]: Internationalizare

**Review by birloiflorian — CHANGES_REQUESTED** [review]

File: src/common/components/filter-modal/FilterModal.tsx (hunk 1 above)
Comment (dragos1195) [reply src/common/components/filter-modal/FilterModal.tsx]: Removed

File: src/common/components/practice-programs-search/PracticeProgramsSearch.tsx (hunk 3 above)
Comment (dragos1195) [reply src/common/components/practice-programs-search/PracticeProgramsSearch.tsx]: Added.

File: src/common/components/practice-programs-search/PracticeProgramsSearch.tsx (hunk 5 above)
Comment (dragos1195) [reply src/common/components/practice-programs-search/PracticeProgramsSearch.tsx]: Fixed.

File: src/common/components/practice-programs-search/PracticeProgramsSearch.tsx (hunk 6 above)
Comment (dragos1195) [reply src/common/components/practice-programs-search/PracticeProgramsSearch.tsx]: Added.

**Review by birloiflorian — APPROVED** [review]

**Pull request merged**